en/tutorials-and-examples/cms/authentication.rst

```diff
@@ -158,7 +158,7 @@ Then add the following::
                 'username' => 'email',
                 'password' => 'password',
             ],
-            'loginUrl' => '/users/login',
+            'loginUrl' => '/users/login', //if your login system is not working then comment out this line
```

Comment:
Can we be more specific on what 'not working' is?

Reply:
I mean login system does not work if i uncomment the below line
`'loginUrl' => '/users/login',`

Even email and passwords are correct but it does not redirect because of invalid url thing, let me add a snapshot for you.

![Screenshot_1](https://user-images.githubusercontent.com/12672971/95292524-0c9ca180-088b-11eb-91b6-7ecabae79e39.png)

Reply:
Ok. It looks like your application is in a sub-directory. The other suggestions in this issue around using Router::url() will resolve this problem better than a comment would.

Reply:
![mark_story](https://user-images.githubusercontent.com/12672971/95425942-8ce00700-095e-11eb-93e4-ecb0b39025c9.png)

Reply:
> I don't know from where it is coming???

From your URL, you are on `/cake4.1.4/admin`, and I would presume that your login form posts to that very endpoint. So either you need to change your form so that it posts to `/users/login` if that is where you want to login, or you need to change the authenticator's `loginUrl` configuration to point to `/admin` if _that_ is where you want to login.

The fact that the base path is missing in the current URL of the error message, indicates that you're possibly not using a recent version of the authentication plugin (check your `composer.json` and make sure to update your dependencies), as the missing base path is [a bug that was fixed](https://github.com/cakephp/authentication/pull/330) quite some time ago.